core/src/test/scala/org/http4s/UriSpec.scala

```diff
@@ -697,4 +697,36 @@ class UriSpec extends Http4sSpec with MustThrownMatchers {
       uri("http://example.com") must_!= uri("http://example.com/")
     }
   }
+
+  "/" should {
+    "encode space as %20" in {
+      uri("http://example.com/") / " " must_== uri("http://example.com/%20")
+    }
+
+    "encode generic delimiters that aren't pchars" in {
+      // ":" and "@" are valid pchars
+      uri("http://example.com") / ":/?#[]@" must_== uri("http://example.com/:%2F%3F%23%5B%5D@")
+    }
+
+    "encode percent sequences" in {
+      uri("http://example.com") / "%2F" must_== uri("http://example.com/%252F")
+    }
+
+    "not encode sub-delims" in {
+      uri("http://example.com") / "!$&'()*+,;=" must_== uri("http://example.com/!$&'()*+,;=")
+    }
+
+    "UTF-8 encode characters" in {
+      uri("http://example.com/") / "ö" must_== uri("http://example.com/%C3%B6")
+    }
+
+    "encode according to charset" in {
+      implicit val cs = Charset.`ISO-8859-1`
+      uri("http://example.com/") / "ö" must_== uri("http://example.com/%F6")      
+    }
+
+    "not make bad URIs" >> forAll { s: String =>
+      Uri.fromString("http://example.com/").map(_ / s) must be_\/-
```

Comment:
Isn't this always going to succeed because the `/` operator will _always_ succeed?